src/resolvers/ticket.resolver.ts

```diff
@@ -1,97 +1,296 @@
+    },
+  },
+
   Query: {
     getAllTickets: async (_: any, arg: any, context: Context) => {
       try {
-        (await checkUserLoggedIn(context))(['superAdmin']);
+        ;(await checkUserLoggedIn(context))([
+          'superAdmin',
+          'admin',
+          'manager',
+          'coordinator',
+          'trainee',
+          'users',
+        ])
+
+        const filterObj: any = {}
 
-        let tickets = await Ticket.find()
+        if (context.role !== 'superAdmin') filterObj.user = context.userId
+
+        const tickets = await Ticket.find(filterObj)
           .populate({
             path: 'user',
-            select: 'email',
+            populate: {
+              path: 'profile',
+              model: 'Profile',
+            },
           })
-          .lean();
-
-        tickets = await Promise.all(await getUserData(tickets));
+          .populate({
+            path: 'replies',
+            populate: {
+              path: 'sender',
+              model: 'User',
+              populate: {
+                path: 'profile',
+                model: 'Profile',
+              },
+            },
+          })
+          .exec()
 
-        return tickets;
+        return tickets
       } catch (error: any) {
-        throw new ApolloError(error.message, '500');
+        throw new ApolloError(error.message, '500')
       }
     },
   },
 
   Mutation: {
     createTicket: async (_: any, args: TicketType, context: Context) => {
       try {
-        const { subject, message }: TicketType = args;
-
-        if (!context.userId) throw new ValidationError('Loggin first');
+        ;(await checkUserLoggedIn(context))([
+          'admin',
+          'coordinator',
+          'manager',
+          'trainee',
+          'user',
+        ])
+        const { subject, message }: TicketType = args
 
         const ticket = await Ticket.create({
           user: context.userId,
           subject,
           message,
-        });
+        })
+
+        const subPayload: any = ticket.toJSON()
+
+        const user = await User.findById(subPayload.user.toString())
+          .populate('profile')
+          .select('email profile id role organization')
 
-        const superAdmin = await User.findOne({ role: 'superAdmin' });
+        subPayload.user = user
 
-        await Notification.create({
+        pubsub.publish('TICKET_CREATED', {
+          ticketCreated: {
+            ...subPayload,
+          },
+        })
+
+        const superAdmin = await User.findOne({ role: 'superAdmin' })
+
+        const notification: any = await Notification.create({
           receiver: superAdmin?._id,
           message: `Ticket has been sent to you. ${ticket._id}`,
           sender: context.userId,
           read: false,
           createdAt: new Date(),
-        });
+        })
+
+        notification.sender = user
+
+        pubsub.publish('SEND_NOTIFICATION_ON_TICKETS', {
+          sendNotsOnTickets: notification,
```

Comment:
Do you publish these events one after another? Or they are sent separately?

Reply:
They one after another